lib/FrameManager.js

```diff
@@ -303,55 +305,88 @@ class Frame {
   }
 
   /**
-   * @param {string} filePath
-   * @return {!Promise<*>}
+   * @param {Object} options
+   * @return {!Promise}
    */
-  async injectFile(filePath) {
-    let contents = await new Promise((resolve, reject) => {
-      fs.readFile(filePath, 'utf8', (err, data) => {
-        if (err) return reject(err);
-        resolve(data);
-      });
-    });
-    contents += `//# sourceURL=` + filePath.replace(/\n/g,'');
-    return this.evaluate(contents);
-  }
+  async addScriptTag(options) {
+    if (typeof options.url === 'string')
+      return this.evaluate(addScriptUrl, options.url);
 
-  /**
-   * @param {string} url
-   */
-  async addScriptTag(url) {
-    return this.evaluate(addScriptTag, url);
+    if (typeof options.path === 'string') {
+      let contents = await readFileAsync(options.path, 'utf8');
+      contents += '//# sourceURL=' + options.path.replace(/\n/g, '');
+
+      return this.evaluate(addScriptContent, contents);
+    }
+
+    if (typeof options.content === 'string')
+      return this.evaluate(addScriptContent, options.content);
+
+    throw new Error('Provide an object with a `url`, `path` or `content` property');
 
     /**
      * @param {string} url
      */
-    function addScriptTag(url) {
+    function addScriptUrl(url) {
       const script = document.createElement('script');
       script.src = url;
       const promise = new Promise(x => script.onload = x);
       document.head.appendChild(script);
       return promise;
     }
+
+    /**
+     * @param {string} content
+     */
+    function addScriptContent(content) {
+      const script = document.createElement('script');
+      script.type = 'text/javascript';
+      script.text = content;
+      document.head.appendChild(script);
+    }
   }
 
   /**
-   * @param {string} url
+   * @param {Object} options
+   * @return {!Promise}
    */
-  async addStyleTag(url) {
-    return this.evaluate(addStyleTag, url);
+  async addStyleTag(options) {
+    if (typeof options.url === 'string')
+      return this.evaluate(addStyleUrl, options.url);
+
+    if (typeof options.path === 'string') {
+      let contents = await readFileAsync(options.path, 'utf8');
+      contents += '//# sourceURL=' + options.path.replace(/\n/g, '');
```

Comment:
@SamVerschueren There's no line comment in css; css sourceurl/sourcemap is put in a block comment:
```js
contents += '/*# sourceURL=' + options.path.replace(/\n/g, '') + '*/';
```

Reply:
Oops, too spoiled using SCSS I guess :).